fuse_examples/segmentation/runner_seg.py

```diff
@@ -0,0 +1,488 @@
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+Created on June 30, 2021
+
+"""
+import os 
+import logging
+from glob import glob
+import random
+import numpy as np
+import pandas as pd
+import matplotlib.pylab as plt
+from pathlib import Path
+from collections import OrderedDict
+
+import torch
+from torch.utils.data import DataLoader
+import torch.optim as optim
+import torch.nn.functional as F
+
+from fuse.data.augmentor.augmentor_toolbox import aug_op_affine_group, aug_op_affine, aug_op_color, aug_op_gaussian, aug_op_elastic_transform
+from fuse.utils.utils_param_sampler import FuseUtilsParamSamplerUniform as Uniform
+from fuse.utils.utils_param_sampler import FuseUtilsParamSamplerRandBool as RandBool
+from fuse.utils.utils_param_sampler import FuseUtilsParamSamplerRandInt as RandInt
+from fuse.utils.utils_gpu import FuseUtilsGPU
+from fuse.utils.utils_logger import fuse_logger_start
+from fuse.data.augmentor.augmentor_default import FuseAugmentorDefault
+from fuse.data.visualizer.visualizer_default import FuseVisualizerDefault
+from fuse.data.dataset.dataset_default import FuseDatasetDefault
+from fuse.models.model_wrapper import FuseModelWrapper
+from fuse.losses.segmentation.loss_dice import DiceBCELoss
+from fuse.losses.segmentation.loss_dice import FuseDiceLoss
+from fuse.losses.loss_default import FuseLossDefault
+from fuse.managers.manager_default import FuseManagerDefault
+from fuse.managers.callbacks.callback_tensorboard import FuseTensorboardCallback
+from fuse.managers.callbacks.callback_metric_statistics import FuseMetricStatisticsCallback
+from fuse.managers.callbacks.callback_time_statistics import FuseTimeStatisticsCallback
+from fuse.data.processor.processor_dataframe import FuseProcessorDataFrame
+# from fuse.metrics.metric_auc_per_pixel import FuseMetricAUCPerPixel
+# from fuse.metrics.segmentation.metric_score_map import FuseMetricScoreMap
+# from fuse.analyzer.analyzer_default import FuseAnalyzerDefault
+from fuse.eval.evaluator import EvaluatorDefault
+from fuse.eval.metrics.segmentation.metrics_segmentation_common import MetricDice, MetricIouJaccard, MetricOverlap, Metric2DHausdorff, MetricPixelAccuracy
+
+from data_source_segmentation import FuseDataSourceSeg
+from seg_input_processor import SegInputProcessor
+
+from unet import UNet
+
+
+def perform_softmax(output):
+    if isinstance(output, torch.Tensor):  # validation
+        logits = output
+    else:  # train
+        logits = output.logits
+    cls_preds = F.softmax(logits, dim=1)
+    return logits, cls_preds
+
+
+SZ = 512
+TRAIN = f'../../data/siim/data{SZ}/train/'
```

Comment:
What is data_dir? How do  I download the data_dir?
Don't expect it to have it in a specific location - ask the user to fill in the path

Reply:
how do you ask the user? 
To leave it blank?